iroha_network/src/lib.rs

```diff
@@ -165,31 +166,33 @@ impl Connection {
     }
 }
 
-impl Stream for Connection {
+impl Iterator for Connection {
     type Item = Vec<u8>;
 
-    fn poll_next(mut self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+    fn next(&mut self) -> Option<Self::Item> {
```

Comment:
Wouldn't it be the problem when we will replace std::net::TcpStream with async_std::net::TcpStream?

Reply:
Then we should change it back to stream when we introduce that change. Here there was a problem with the fact that we wrapped synchronous blocking code with async stream. And it just did not execute, I think the async_std scheduler was blocked by this call and could not schedule other tasks properly.

Reply:
ok, ty